### PR TITLE
Add Methods Interfaces

### PR DESCRIPTION
**Pull request by fzimmermann89**
Some Remarks:

- There is a lot of noise, as I tried to merge all the changes done in parallel...
- The order of the imports in the inferfaces/__init__.py is important
- FloatTensor etc is not useful. A torch.tensor with dtype=float32 is not a FloatTensor according to mypy and pylance.

### CODE REVIEW

File: src/inhomcorr/interfaces/bias_estimator_interface.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"""Interfaces for bias field estimators."""
+from abc import ABC
+from abc import abstractmethod
+
+from .mr_data_interface import ImageData
```
Comment (schuenke) [inline src/inhomcorr/interfaces/bias_estimator_interface.py:5]: this should be `from inhomcorr.interfaces import ImageData`, no?!

**Comment by schuenke** [comment]
We did agree on the import style that we used for example in https://github.com/ckolbPTB/InhomCorr/blob/main/src/inhomcorr/mrsig/flash.py, no?

```python
from inhomcorr.interfaces import ImageData
from inhomcorr.interfaces import MRParam
from inhomcorr.interfaces import MRSig
from inhomcorr.interfaces import QMRIData
```

I recommend that we use it from the beginning on. Would be great if somebody could adjust it in all changed files or this PR. I have no internet on my laptop and thus cannot do it myself... sorry...

**Comment by fzimmermann89** [comment]
Ok, my observations: 
Either we follow the convention also insides the interfaces module (as @schuenke  requested), then the order of the import statements in the __init__.py is important!
The ImageData and QMRIData has to be imported before the modules depending on it. 
So we need isort skip statements, otherwise the imports starting with the letter 'b' will be sorted before it..

Or we deviate inside the interfaces from this convention, as @johannesmayer suggested.

Not sure what will work best in the long run, for now I did @schuenke 's suggestions, we may want to switch back later.

**Comment by fzimmermann89** [comment]
And @schote : The FloatTensor stuff just does not help, a tensor of dtype float is not a FloatTensor etc.... :-(

**Comment by fzimmermann89** [comment]
Also, to whoever is gonna merge this: I would not mind if you squash-merge it.

This PR contains way to many useless commits that can all be squashed to one.

**Review by schote — APPROVED** [review]

**Comment by schote** [comment]
> > And @schote : The FloatTensor stuff just does not help, a tensor of dtype float is not a FloatTensor etc.... :-(
> 
> I had the same feeling earlier, what do you think, should we just use `tensor.Tensor` then?

I see you have already changed the data types 👍🏼

**Pull request merged**